### PR TITLE
feat(VSnackbar): add new properties and improve md2 spec

### LINKED ISSUE #5865: [Bug Report] Re-using a snackbar with different timeout does not work
### Versions and Environment
**Vuetify:** 1.3.12
**Vue:** 2.5.20
**Browsers:** Chrome 70.0.3538.110
**OS:** Mac OS 10.14.1

### Steps to reproduce
In codepen:
1. Press "show snackbar"
2. Press one of the other two buttons within 3 seconds

### Expected Behavior
Snackbar will stay for 10 sec or forever, dependent on button pressed

### Actual Behavior
Snackbar disappears 3 seconds after initial display

### Reproduction Link
<a href="https://codepen.io/everhardt/pen/pqJvNM" target="_blank">https://codepen.io/everhardt/pen/pqJvNM</a>

### Other comments
The timeout parameter can now only be used when the v-model of the snackbar is set to false. Reusing the snackbar with a different timeout cannot easily be done when it is currently shown, as
```
this.show = false;
this.timeout = 0;
this.show = true;
```
will not trigger a refresh. Only
```
this.show = false;
Vue.nextTick(function () {
  // DOM updated
  this.timeout = 0;
  this.show = true;
})
```

will.

### LINKED ISSUE #10935: [Feature Request] add Outline and Text properties to snackbar component as in alert
### Problem to solve
this will improve snackbar UI

### Proposed solution
simply add Outline and Text properties

### PR DESCRIPTION
**Pull request by johnleider**
added support for elevation, rounded, and theme

## Description
Changed the default value for **timeout** from `6000` to `5000` to match specification. Changed the value to disable the timeout from `0` to `-1`. The **timeout** property now accepts strings and is converted using `Number`.

Added new scoped slot that provides attributes for `v-btn` styling.
```vue
<template>
  <v-snackbar>
      This item already has the label "travel". You can add a new label. You can add a new label.

      <template v-slot:action="{ attrs }">
        <v-btn
          v-bind="attrs"
          text
        >
          Add a new label
        </v-btn>
      </template>
    </v-snackbar>
</template>
```
Added and deprecated multiple SASS variables.
* $snackbar-font-size
* $snackbar-btn-margin-left
* $snackbar-btn-margin-right
* $snackbar-content-first-btn-margin
* $snackbar-corner-offset
* $snackbar-wrapper-max-width
* $snackbar-wrapper-min-width

Added the following new props:

* centered - centers the snackbar on the x and y axis
* content-class
* dark
* elevation
* light
* outlined
* shaped
* rounded
* text
* transition

Also refactored all tests.

## Motivation and Context
Improve `v-snackbar` heading into LTS.

fixes #10935
fixes #5865

## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
<template>
  <div class="ma-12 pa-12">
    <v-btn @click="value = !value">Click me</v-btn>

    <v-snackbar
      :value="value"
      timeout="-1"
      shaped
      top
      multi-line
      left
    >
      This item already has the label "travel". You can add a new label. You can add a new label.

      <template v-slot:action="{ attrs }">
        <v-btn
          v-bind="attrs"
          text
        >
          Add a new label
        </v-btn>

        <v-btn
          class="ml-2"
          icon
        >
          <v-icon>$close</v-icon>
        </v-btn>
      </template>
    </v-snackbar>

    <v-snackbar
      :value="value"
      timeout="-1"
      color="error"
      vertical
      centered
      outlined
    >
      This item already has the label.

      <template v-slot:action="{ attrs }">
        <v-btn
          v-bind="attrs"
          text
        >
          Add a new label
        </v-btn>

        <v-btn
          class="ml-2"
          icon
        >
          <v-icon>$close</v-icon>
        </v-btn>
      </template>
    </v-snackbar>

    <v-snackbar
      :value="value"
      timeout="-1"
      color="primary"
      vertical
      text
    >
      This item already has the label "travel". You can add a new label.

      <template v-slot:action="{ attrs, on }">
        <v-btn
          v-bind="attrs"
          text
        >
          Add a new label
        </v-btn>

        <v-btn
          class="ml-2"
          icon
          v-on="on"
        >
          <v-icon>$close</v-icon>
        </v-btn>
      </template>
    </v-snackbar>

    <!-- <v-navigation-drawer
      :value="value"
      app
    ></v-navigation-drawer> -->
  </div>
</template>

<script>
  export default {
    data: () => ({
      value: true,
    }),

    mounted () {
      // setInterval(() => {
      //   this.value = !this.value
      // }, 3000)
    },
  }
</script>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by johnleider** [comment]
Requesting **Initial** comments from @vuetifyjs/contributors @vuetifyjs/core-team

**Comment by nekosaur** [comment]
Haven't looked at the code yet, but this

> Changed the value to disable the timeout from 0 to -1.

is a breaking change unless 0 still works.

**Comment by KaelWD** [comment]
> centers the snackbar on the x and y axis

Why would you ever want a snackbar directly in the middle of the screen?

File: packages/vuetify/src/components/VSnackbar/_variables.scss [hunk 1]
```diff
@@ -1,23 +1,40 @@
...
-$snackbar-left: 8px !default;
-$snackbar-multi-line-content-min-height: 68px !default;
-$snackbar-right: 8px !default;
-$snackbar-top: 8px !default;
+$snackbar-bottom: 0 !default;
+$snackbar-btn-padding: 0 8px !default;
+$snackbar-color: hsla(0, 0%, 100%, .87) !default;
+$snackbar-content-font-size: map-deep-get($headings, 'body-2', 'size') !default;
+$snackbar-content-font-weight: map-deep-get($headings, 'body-2', 'weight') !default;
+$snackbar-content-font-weight: map-deep-get($headings, 'body-2', 'weight') !default;
```
Comment (KaelWD) [inline packages/vuetify/src/components/VSnackbar/_variables.scss:11]: Duplicated

**Comment by johnleider** [comment]
Requesting **Final** comments from @vuetifyjs/contributors @vuetifyjs/core-team

File: packages/vuetify/src/components/VSnackbar/VSnackbar.ts [hunk 2]
```diff
@@ -60,64 +104,113 @@ export default mixins(
...
+          [this.contentClass]: true,
+        },
+        attrs: {
+          role: 'status',
+          'aria-live': 'polite',
+        },
+      }, [getSlot(this)])
+    },
+    genWrapper () {
+      const fn = this.hasBackground
```
Comment (sh7dm) [inline packages/vuetify/src/components/VSnackbar/VSnackbar.ts:163]: `setColor` or another non-generic name?

**Pull request merged**